test/Conversion/TorchToLinalg/view.mlir

```diff
@@ -73,4 +151,137 @@ func.func @torch.aten.view$expandInferredDim(%arg0: !torch.vtensor<[2,6],f32>) -
     %0 = torch.prim.ListConstruct %int3, %int2, %int-1 : (!torch.int, !torch.int, !torch.int) -> !torch.list<int>
     %1 = torch.aten.view %arg0, %0 : !torch.vtensor<[2,6],f32>, !torch.list<int> -> !torch.vtensor<[3,2,2],f32>
     return %1 : !torch.vtensor<[3,2,2],f32>
+}
+
+// -----
+
+// CHECK-LABEL: func.func @torch.aten.view$expandWithOnes(
+// CHECK-SAME:    %[[ARG:.*]]: !torch.vtensor<[8],f32>) -> !torch.vtensor<[?,?,8,1,?],f32> {
+// CHECK:     %[[BUILTIN_TENSOR:.*]] = torch_c.to_builtin_tensor %[[ARG]] : !torch.vtensor<[8],f32> -> tensor<8xf32>
+// CHECK:   %[[EXPAND:.*]] = tensor.expand_shape %[[BUILTIN_TENSOR]] {{\[\[}}0, 1, 2, 3, 4]] : tensor<8xf32> into tensor<1x1x8x1x1xf32>
+// CHECK:  %[[CAST:.*]] = tensor.cast %[[EXPAND]] : tensor<1x1x8x1x1xf32> to tensor<?x?x8x1x?xf32>
+// CHECK: %[[BUILTIN_TENSOR_CAST:.*]] = torch_c.from_builtin_tensor %[[CAST]] : tensor<?x?x8x1x?xf32> -> !torch.vtensor<[?,?,8,1,?],f32>
+// CHECK: return %[[BUILTIN_TENSOR_CAST]] : !torch.vtensor<[?,?,8,1,?],f32>
+
+func.func @torch.aten.view$expandWithOnes(%arg0: !torch.vtensor<[8],f32>) -> !torch.vtensor<[?,?,8,1,?],f32> {
+    %int8 = torch.constant.int 8
+    %int1 = torch.constant.int 1
+    %int-1 = torch.constant.int -1
+    %0 = torch.prim.ListConstruct %int-1, %int-1, %int8, %int1, %int-1 : (!torch.int, !torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
+    %1 = torch.aten.view %arg0, %0 : !torch.vtensor<[8],f32>, !torch.list<int> -> !torch.vtensor<[?,?,8,1,?],f32>
```

Comment:
Is this a legal view op? My understanding was that you can only have one inferred (`-1`) dimension size.
(IIRC reshape lowers to view: https://pytorch.org/docs/stable/generated/torch.reshape.html#torch.reshape)

Reply:
I'm not sure. I don't have a good understanding of how the classic pytorch '-1' and dynamic shape '?', as created by dynamo, are related.  But this IR does look a bit confused..